presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -183,13 +183,23 @@ public ActualProperties visitEnforceSingleRow(EnforceSingleRowNode node, List<Ac
         public ActualProperties visitAssignUniqueId(AssignUniqueId node, List<ActualProperties> inputProperties)
         {
             ActualProperties properties = Iterables.getOnlyElement(inputProperties);
+
+            ImmutableList.Builder<LocalProperty<Symbol>> localPropertiesBuilder = ImmutableList.builder();
+            localPropertiesBuilder.addAll(properties.getLocalProperties());
+            localPropertiesBuilder.add(new GroupingProperty<>(ImmutableList.of(node.getIdColumn())));
+            node.getSource().getOutputSymbols().stream()
+                    .forEach(column -> localPropertiesBuilder.add(new ConstantProperty<>(column)));
```

Comment:
I wouldn't bother with those for now. It makes reasoning more difficult and won't be used anyway.

Reply:
@sopel39 Yes, it has been quite a challenge to reason through all of these changes. I'm getting better at it now, but it is still very hard. My preference would be to produce strongest possible properties, which in this case seems to be `G(unique), C(a), C(b),...` as it will allow adding functionality later without revisiting this complicated topic. :-)

For example, theoretically, `G(unique), C(a), C(b),...` properties allow streaming aggregation to compare only `unique` columns, but we need to find a way to propagate that level of detail to the execution.